src/Workspaces/Core/Portable/PatternMatching/AllLowerCamelCaseMatcher.cs

```diff
@@ -20,23 +21,22 @@ internal partial class PatternMatcher
         /// a candidate using CamelCase matching. i.e. this code is responsible for finding the
         /// match between "cofipro" and "CodeFixProvider". 
         /// </summary>
-        private struct AllLowerCamelCaseMatcher
+        private ref struct AllLowerCamelCaseMatcher
         {
             private readonly bool _includeMatchedSpans;
             private readonly string _candidate;
-            private readonly ArrayBuilder<TextSpan> _candidateHumps;
```

Comment:
this was annoying.  can't point to a struct by ref as a field.  So this had to be a parameter passed to all methods instead.

Reply:
**I'm fine with the current approach for this PR.** The available alternative is placing the non-copyable value in heap storage instead of stack storage, e.g. `StrongBox<TemporaryArray<TextSpan>>` or `Tuple<>`/`class`, depending on the number of values that need to be stored together. The two main downsides of this approach is it forces an additional allocation (perhaps not bad if the value can be used many times) and it forces all callers to change signatures.